table/policy.go

```
@@ -3337,14 +3337,26 @@ func (r *RoutingPolicy) DeleteStatement(st *Statement, all bool) (err error) {
 	return err
 }
 
-func (r *RoutingPolicy) ReplaceStatement(st *Statement) (err error) {
+func (r *RoutingPolicy) ReplaceStatement(st *Statement, all bool) (err error) {
 	r.mu.Lock()
 	defer r.mu.Unlock()
 
+	for _, c := range st.Conditions {
+		if err = r.validateCondition(c); err != nil {
+			return
```

I think we should return the error here